crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -2025,6 +2025,35 @@ impl<'w> EntityWorldMut<'w> {
         )
     }
 
+    /// Triggers the given [`Event`], which will run any [`Observer`]s watching for it.
+    ///
+    /// [`Observer`]: crate::observer::Observer
+    #[track_caller]
+    pub fn trigger<'a, E: Event<Trigger<'a>: Default>>(&mut self, mut event: E) {
+        self.world_scope(|world| {
+            world.trigger_ref_with_caller(
+                &mut event,
+                &mut <E::Trigger<'_> as Default>::default(),
+                MaybeLocation::caller(),
+            );
+        });
+    }
+
+    /// Triggers the given [`Event`] using the given [`Trigger`], which will run any [`Observer`]s watching for it.
+    ///
+    /// [`Trigger`]: crate::event::Trigger
+    /// [`Observer`]: crate::observer::Observer
+    #[track_caller]
+    pub fn trigger_with<'a, E: Event<Trigger<'a>: Send + Sync>>(
+        &mut self,
+        mut event: E,
+        mut trigger: E::Trigger<'a>,
+    ) {
+        self.world_scope(|world| {
+            world.trigger_ref_with_caller(&mut event, &mut trigger, MaybeLocation::caller());
+        });
+    }
```

Comment:
It feels confusing that these are on a `EntityWorldMut` but don't use it at all.

Reply:
I'm not against removing them. The thought was to make it easier to trigger events _during_ entity construction, but in practice I think that is unnecessary.

Reply:
Removed!